src/main/java/io/github/dgroup/enumerable4j/Enumerable.java

```diff
@@ -47,69 +47,89 @@
 public interface Enumerable<X> extends Collection<X> {
 
     /**
-     * Passes each element of the collection to the given block.
+     * Passes each element of the collection to the each given function.
      * If no predicate (null) is given, then true is returned instead.
-     * @param prd The predicate to match each element.
-     * @return The true if the block never returns false or nil.
+     * @param prd The array of functions to match each element.
+     * @return True if the functions never return false or nil.
      */
-    default boolean all(Predicate<X> prd) {
-        final boolean match;
-        if (prd == null) {
-            match = true;
-        } else {
-            match = this.stream().allMatch(prd);
+    default boolean all(Predicate<X>... prd) {
+        boolean match = true;
+        if (prd != null) {
+            for (int idx = 0; match && idx < prd.length; ++idx) {
+                if (prd[idx] != null) {
+                    match = this.stream().allMatch(prd[idx]);
+                }
+            }
         }
         return match;
     }
 
     /**
-     * Passes at least one element of the collection to the given block.
+     * Passes at least one element of the collection to the each given function.
      * If no predicate (null) is given, then true is returned instead.
-     * @param prd The predicate to match at least one element.
-     * @return The true if the block never returns false or nil.
+     * @param prd The array of functions to match at least one element.
+     * @return True if the functions never return false or nil.
      */
-    default boolean any(Predicate<X> prd) {
+    default boolean any(Predicate<X>... prd) {
```

Comment:
please use `Predicate<T> first, Predicate<T> ...other` everywhere.

Reply:
what should be if any of the predicates is null?
should it throw any exception? or should it return true?

Reply:
for example, what should the next method return?
```java
src.any( pred , null, null, null);
```

Reply:
i think let's just skip the `null` based predicates

Reply:
So, if all arguments are ‘null’ the result is ‘true’. right?

Reply:
@dgroup , please check the new implementation. 
If it’s ok merge the changes please. 
The implementation of `find`, `next` etc methods should be discussed later